### PR TITLE
allow overriding release name

### PR DESCRIPTION
**Pull request by vhdirk**
Fixes https://github.com/eclipse/packages/issues/430

Signed-off-by: Dirk Van Haerenborgh <dirk.vanhaerenborgh@aloxy.io>

### CODE REVIEW

**Comment by calohmn** [comment]
@vhdirk The goal of #252 is for the _subject alternative names_ of the example certificates to always match the Hono service names, so that we can enable hostname verification in the clients. With the release name being part of the service name, matching names cannot be guaranteed. Therefore the discussion in #252 lead to the plan to remove the release name from the Hono service names, meaning service names are fixed (see https://github.com/eclipse/packages/issues/252#issuecomment-883204654). 

In your PR here, this approach is not taken. Can you explain what the relation to #252 is?

Just looking at the PR changes itself, I see that these would harmonize K8s resource naming with the approach taken in the ditto chart. This would mean that in the cloud2edge chart, service names would then be
```
c2e-hono-adapter-amqp
c2e-hono-service-auth
c2e-ditto-gateway
c2e-ditto-nginx
```
and not 
```
c2e-adapter-amqp
c2e-service-auth
c2e-ditto-gateway
c2e-ditto-nginx
```
any more, making the naming clearer. This will need changes in the cloud2edge chart as well though (e.g. in scripts invoking the services). Therefore we have to think about versioning and compatibility here. (To stay compatible with the cloud2edge chart, we would even have to make this change here a major version jump. I guess we could consider adapting the cloud2edge chart Hono requirements version range though (to ~2.1.0), requiring only a minor version jump here.)

**Comment by vhdirk** [comment]
When asking on gitter if there was an option to override the release name, I was pointed to #252, which is why I referenced it.
I chose not to remove the release name entirely, since we internally also have a service called 'adapter'. The removal of release names from Hono would make it too ambiguous. And we also like the parity with ditto.

I can't imagine removing the prefix altogether would be the only option with regard to hostname verification?

**Comment by sophokles73** [comment]
> When asking on gitter if there was an option to override the release name, I was pointed to https://github.com/eclipse/packages/issues/252,

I am sorry, that was my fault. I had simply thought that we could kill two birds with one stone. However, I guess this PR has value on its own, even if it does not address the host name verification problem itself. I agree with @calohmn that we need be transparent about backward compatibility (or the lack thereof) by means of choosing a proper version number increase.

@calohmn apart from the version and corresponding release notes entry, is there anything else that needs to be changed?

> I can't imagine removing the prefix altogether would be the only option with regard to hostname verification?

Probably not, I am currently thinking in the direction to (dynamically) create the demo certificates by means of cert-manager, i.e. making cert-manager a pre-requisite for the Hono chart when using demo certs.

File: charts/hono/templates/hono-service-device-registry-base/hono-service-device-registry-post-install-job.yaml [hunk 1]
```diff
@@ -48,6 +48,6 @@ spec:
       volumes:
       - name: post-install-data
         configMap:
-          name: {{ printf "%s-post-install-device-registry-conf" .Release.Name }}
+          name: {{ include "hono.fullname" . }}"-post-install-device-registry-conf"
```
Comment (sophokles73) [inline charts/hono/templates/hono-service-device-registry-base/hono-service-device-registry-post-install-job.yaml:51]: this produces `"hono\"-post-install-device-registry-conf\"` which causes the deployment test to fail

I suggest to use
```yaml
name: {{ printf "%s-post-install-device-registry-conf" ( include "hono.fullname" . ) }}
```

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: charts/hono/templates/hono-service-device-registry-base/hono-service-device-registry-post-install-job.yaml (hunk 1 above)
Comment (vhdirk) [reply charts/hono/templates/hono-service-device-registry-base/hono-service-device-registry-post-install-job.yaml]: fixed it for al occurrences of the same thing

File: charts/hono/Chart.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ name: hono
 description: |
   Eclipse Hono™ provides remote service interfaces for connecting large numbers of IoT devices to a back end and
   interacting with them in a uniform way regardless of the device communication protocol.
-version: 2.1.5
```
Comment (calohmn) [inline charts/hono/Chart.yaml:18]: Regarding cloud2edge compatibility, we have 2 options here:
- use version `3.0.0` here. That means the current  cloud2edge chart won't reference it (`^2.1.0` used there now).
- use version `2.2.0` here and first create a new cloud2edge chart version that only changes the Hono dependency requirement from `^2.1.0` to `~2.1.0`.

I would tend towards the 2nd option here. Having just the patch-level version range in the cloud2edge chart offers more flexibility regarding Hono chart changes affecting the cloud2edge chart, without having to do a major version jump of the Hono chart each time.
@sophokles73 WDYT?

File: charts/hono/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ name: hono
...
+version: 2.1.6
```
Comment (calohmn) [inline charts/hono/Chart.yaml:18]: @vhdirk Can you also add an entry in the `Release Notes` section of the `README.md`?

**Review by sophokles73 — CHANGES_REQUESTED** [review]

File: charts/hono/templates/_helpers.tpl [hunk 4]
```diff
@@ -102,7 +102,7 @@ The scope passed in is expected to be a dict with keys
 - "component": the value to use for the "app.kubernetes.io/component" label
 */}}
 {{- define "hono.metadata" -}}
-name: {{ printf "%s-%s" .dot.Release.Name .name | quote }}
+name: {{ printf "%s-%s" (include "hono.fullname" .dot ) .name | quote }}
```
Comment (calohmn) [inline charts/hono/templates/_helpers.tpl:105]: It looks like there are still some more replacements like this needed:
- the pod list in `NOTES.txt`
- data-grid server name in `hono-data-grid.xml`
- host entries in `qdrouterd.json`
- `HTTP_BASE_URL` in `add_example_data_device_registry.sh`
- `{{ .Release.Name }}-kafka-example-keys` in `values.yaml`

File: charts/hono/templates/example-data-grid/statefulset.yaml [hunk 5]
```diff
@@ -74,5 +74,5 @@ spec:
       volumes:
       - name: conf
         configMap:
-          name: {{ printf "%s-%s-conf" .Release.Name $args.name }}
+          name: {{ printf "%s-%s-conf" (include "hono.name" .) $args.name }}
```
[the remark this answers is not in the document]
Comment (vhdirk) [reply charts/hono/templates/example-data-grid/statefulset.yaml:77]: Not really, I took inspiration from the ditto helm chart, that is all. I'll change all occurrences

File: charts/hono/Chart.yaml (hunk 2 above)
Comment (vhdirk) [reply charts/hono/Chart.yaml:18]: I also gravitate more towared the second option here.

File: charts/hono/Chart.yaml (hunk 3 above)
Comment (vhdirk) [reply charts/hono/Chart.yaml]: done. Changed to v2.2.0 as per https://github.com/eclipse/packages/pull/416#discussion_r1001436531

File: charts/hono/templates/_helpers.tpl (hunk 4 above)
Comment (vhdirk) [reply charts/hono/templates/_helpers.tpl:105]: done

File: charts/hono/templates/example-data-grid/statefulset.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (vhdirk) [reply charts/hono/templates/example-data-grid/statefulset.yaml]: fixed

**Comment by sophokles73** [comment]
@vhdirk with all the discussion around whether this PR actually helps with fixing the SAN verification problem, I have lost track of what this PR actually tries to achieve. Would you mind creating a dedicated issue describing the lacking functionality and/or problems with the currently implemented naming strategy? We can then better assess, if your PR goes in the right direction or should better take another route to achieving the goal.
Sorry for the inconvenience but I guess the discussion will become more focused this way.

**Comment by vhdirk** [comment]
@sophokles73 no problem, that's perfectly understandable. I've created https://github.com/eclipse/packages/issues/430 and linked this PR to it

**Comment by sophokles73** [comment]
@vhdirk can we close this PR?

**Comment by calohmn** [comment]
Closing this PR as this feature was merged via #469.

**Pull request closed without merging**